### PR TITLE
Added containerized local website authoring

### PR DESCRIPTION
**Pull request by somaritane**
Added support for containerized local website authoring and serving,
without the need for external Ruby/Jekyll dependencies.

Serving is done via the `serve` make target (default), with the use of `jekyll/jekyll` docker image.
Target requires a GitHub PAT token with the 'public_repo' scope assigned to `JEKYLL_GITHUB_TOKEN` environment variable, for GitHub repository metadata retrieval during generation.

Fixes: #483

Signed-off-by: Timofey Ilinykh <timofey.ilinykh@absa.africa>

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -259,11 +259,12 @@ GEM
...
+  x86_64-linux-musl
 
 DEPENDENCIES
   github-pages (~> 214)
+  jekyll-github-metadata
   webrick (~> 1.7)
 
 BUNDLED WITH
-   2.2.17
+   2.2.2
```
Comment (kuritka) [inline Gemfile.lock:270]: isn't 2.2.2 lower than 2.2.17 ?

Comment (somaritane) [reply Gemfile.lock:270]: it is, but this is the version of bundler in `jekyll/jekyll` image (https://hub.docker.com/r/jekyll/jekyll)

**Review by kuritka — APPROVED** [review]
lgtm

**Review by ytsarev — COMMENTED** [review]
Shouldn't it be supported by parallel documentation update in CONTRIBUTING?

**Comment by somaritane** [comment]
> Shouldn't it be supported by parallel documentation update in CONTRIBUTING?

@ytsarev it certainly should, but this PR merges into [/gh-pages](https://github.com/k8gb-io/k8gb/commits/gh-pages) branch. CONTRIBUTING.md will be updated with separate PR.

**Comment by k0da** [comment]
Do we still need Gemfile?

**Comment by ytsarev** [comment]
> > Shouldn't it be supported by parallel documentation update in CONTRIBUTING?
> 
> @ytsarev it certainly should, but this PR merges into [/gh-pages](https://github.com/k8gb-io/k8gb/commits/gh-pages) branch. CONTRIBUTING.md will be updated with separate PR.

yeah, that's why I meant by parallel. It will not hurt to review this one together with documentation update to `master` branch so it will not slip

**Comment by somaritane** [comment]
> > > Shouldn't it be supported by parallel documentation update in CONTRIBUTING?
> > 
> > 
> > @ytsarev it certainly should, but this PR merges into [/gh-pages](https://github.com/k8gb-io/k8gb/commits/gh-pages) branch. CONTRIBUTING.md will be updated with separate PR.
> 
> yeah, that's why I meant by parallel. It will not hurt to review this one together with documentation update to `master` branch so it will not slip

@ytsarev, makes sense, updated CONTRIBUTING documentation in #494, please check

**Comment by somaritane** [comment]
> Do we still need Gemfile?

@k0da yes, this is still needed for local website generation in `jekyll/jekyll` container. While GitHub pages infra has all the necessary dependencies preinstalled, we need to re-create GH pages build environment. And dependencies for that can be easily installed during run time using bundler and Gemfile.

**Review by ytsarev — CHANGES_REQUESTED** [review]
I hit locally. What am I doing wrong? 
```
 make serve
Building and serving k8gb.io website at http://localhost:4000 ⌛ ...
Unable to find image 'jekyll/jekyll:4.2.0' locally
4.2.0: Pulling from jekyll/jekyll
cbdbe7a5bc2a: Pull complete
aa8ae8202b42: Pull complete
b21786fe7c0d: Pull complete
68296e6645b2: Pull complete
6b1c37303e2d: Pull complete
88c5c9c15920: Pull complete
Digest: sha256:de110e0fdde342de0cb6fd5c63f64934b6ce25d0dd826c67113b3a5a3370784e
Status: Downloaded newer image for jekyll/jekyll:4.2.0
ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux-musl]
Traceback (most recent call last):
	12: from /usr/gem/bin/jekyll:23:in `<main>'
	11: from /usr/gem/bin/jekyll:23:in `load'
	10: from /usr/gem/gems/jekyll-4.2.0/exe/jekyll:11:in `<top (required)>'
	 9: from /usr/gem/gems/jekyll-4.2.0/lib/jekyll/plugin_manager.rb:52:in `require_from_bundler'
	 8: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler.rb:149:in `setup'
	 7: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/runtime.rb:20:in `setup'
	 6: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/runtime.rb:91:in `block in definition_method'
	 5: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/definition.rb:227:in `requested_specs'
	 4: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/definition.rb:245:in `specs_for'
	 3: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/definition.rb:175:in `specs'
	 2: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/spec_set.rb:80:in `materialize'
	 1: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/spec_set.rb:80:in `map!'
/usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/spec_set.rb:86:in `block in materialize': Could not find concurrent-ruby-1.1.8 in any of the sources (Bundler::GemNotFound)
make: *** [serve] Error 1
```

**Comment by somaritane** [comment]
> I hit locally.
> 
> ```
>  make serve
> Building and serving k8gb.io website at http://localhost:4000 ⌛ ...
> Unable to find image 'jekyll/jekyll:4.2.0' locally
> 4.2.0: Pulling from jekyll/jekyll
> cbdbe7a5bc2a: Pull complete
> aa8ae8202b42: Pull complete
> b21786fe7c0d: Pull complete
> 68296e6645b2: Pull complete
> 6b1c37303e2d: Pull complete
> 88c5c9c15920: Pull complete
> Digest: sha256:de110e0fdde342de0cb6fd5c63f64934b6ce25d0dd826c67113b3a5a3370784e
> Status: Downloaded newer image for jekyll/jekyll:4.2.0
> ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux-musl]
> Traceback (most recent call last):
> 	12: from /usr/gem/bin/jekyll:23:in `<main>'
> 	11: from /usr/gem/bin/jekyll:23:in `load'
> 	10: from /usr/gem/gems/jekyll-4.2.0/exe/jekyll:11:in `<top (required)>'
> 	 9: from /usr/gem/gems/jekyll-4.2.0/lib/jekyll/plugin_manager.rb:52:in `require_from_bundler'
> 	 8: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler.rb:149:in `setup'
> 	 7: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/runtime.rb:20:in `setup'
> 	 6: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/runtime.rb:91:in `block in definition_method'
> 	 5: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/definition.rb:227:in `requested_specs'
> 	 4: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/definition.rb:245:in `specs_for'
> 	 3: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/definition.rb:175:in `specs'
> 	 2: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/spec_set.rb:80:in `materialize'
> 	 1: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/spec_set.rb:80:in `map!'
> /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.2.2/lib/bundler/spec_set.rb:86:in `block in materialize': Could not find concurrent-ruby-1.1.8 in any of the sources (Bundler::GemNotFound)
> make: *** [serve] Error 1
> ```
> 
> What am I doing wrong?

@ytsarev this looks like your corp env issue.

**Comment by ytsarev** [comment]
@somaritane it doesn't *look* so but it indeed is. Also it is *our* corp issue :D

**Review by ytsarev — APPROVED** [review]

**Pull request merged**